backend/graphql/handler.ts

```diff
@@ -1,25 +1,68 @@
 import { executeQuery } from "../psql/queryExecute";
-import { QUERY_GET_ALL_DAOS } from "../psql/query";
-import { DaoType } from "./types";
-import { DaoItemType } from "../types";
-const { GraphQLObjectType, GraphQLList } = require("graphql");
+import { QUERY_GET_DAOS_COUNT, QUERY_PAGINATED_DAOS } from "../psql/query";
+import { DaosAndPageInfoType } from "./types";
+import { DaoItemType, DaoArgType, DaoAndPageResType } from "../types";
+const { GraphQLObjectType, GraphQLInt, GraphQLNonNull } = require("graphql");
+
+const validateRes = (totalDaosRes: any, paginateRes: any, args: DaoArgType) => {
+	return (
+		totalDaosRes &&
+		totalDaosRes.rows &&
+		paginateRes &&
+		paginateRes.rows &&
+		paginateRes.rows.length &&
+		args
+	);
+};
+
+const getDaoPageRes = (
+	totalDaos: number,
+	Daos: [DaoItemType],
+	args: DaoArgType
+): DaoAndPageResType => {
+	const pageNumber = args.pageNumber; // page number = 1..n
+	const pageSize = args.pageSize;
+	const hasNext = !(pageNumber * pageSize >= totalDaos); // check if user's total page iterated is >= totalDaos
+	const hasPrev = !(pageNumber <= 1); // check if user is on first page
+	return {
+		Daos,
+		pageInfo: {
+			hasNext,
+			hasPrev,
+		},
+	};
+};
 
 export const queryRoot = new GraphQLObjectType({
 	name: "Query",
 	fields: () => ({
-		Daos: {
-			type: new GraphQLList(DaoType),
-			description: "Dao Records",
-			resolve: async () => {
-				const res = await executeQuery(QUERY_GET_ALL_DAOS);
-				if (res && res.rows && res.rows.length) {
-					res.rows.map((item: DaoItemType) => {
+		DaoAndPage: {
+			type: DaosAndPageInfoType,
+			description: "Returns Dao Records and Page Info",
+			args: {
+				pageNumber: {
+					type: new GraphQLNonNull(GraphQLInt),
+					description: "Page number as an argument",
+				},
+				pageSize: {
+					type: new GraphQLNonNull(GraphQLInt),
+					description: "Offset as an argument.",
+				},
+			},
+			resolve: async (_: object, args: DaoArgType) => {
+				const totalDaosRes = await executeQuery(QUERY_GET_DAOS_COUNT);
+				const paginateRes = await executeQuery(
+					QUERY_PAGINATED_DAOS(args.pageNumber, args.pageSize)
```

Comment:
why uppercase? Uppercase should be only for constants

Reply:
Done